File: src/Web/CombatAnalysis.EnhancedWebApp/CombatAnalysis.EnhancedWebApp.Server/Models/GameLogs/CombatPlayerData/ResourceRecoveryModel.cs
namespace CombatAnalysis.EnhancedWebApp.Server.Models.GameLogs.CombatPlayerData;

public class ResourceRecoveryModel
{
    public string Id { get; set; }

    public int GameSpellId { get; set; }

    public string Spell { get; set; }

    public int Value { get; set; }

    public string Time { get; set; }

    public UnitModel Unit { get; set; } = new();

    public UnitModel Target { get; set; } = new();

    public int ModificationType { get; set; }

    public string UnitId { get; set; }
}

File: src/Web/CombatAnalysis.EnhancedWebApp/combatanalysis.enhancedwebapp.client/src/features/gameLogs/types/ResourceRecoveryModel.ts
import type { UnitModel } from './UnitModel';

export type ResourceRecoveryModel = {
    id: number;
    spell: string;
    value: number;
    time: string;
    unit: UnitModel;
    target: UnitModel;
    modificationType: number;
    combatPlayerId: number;
}